### ISSUE TITLE
Bootstrapping code-server setup

### ISSUE THREAD

**Issue by Zireael** [cdr/code-server#1183]
Thought I'd share how I'm bootstrapping setup of code-server in a docker image, with the help of [code-settings-sync](https://github.com/shanalikhan/code-settings-sync) extension.
Originally posted here: https://github.com/shanalikhan/code-settings-sync/issues/1050#issuecomment-554802122

I know I could do it  with `code-server --install-extension xxx.yyy` or using *.vsix package files, but this is the most straightforward way. One downside is, I can't find a way to initialize 'first run' of code-server during Dockerfile build so that code-settings-sync installs all extensions during build time instead of dockerimage first bootup for user interaction.

Would it be possible to add a command line for code-server init/bootstrap like this? ` RUN code-server --first-run-bootstrap`

---
I'm doing settings sync on first run like this in my Linux docker file with online version of VSCode called [code-server](https://github.com/cdr/code-server). If you change `code-server` in script below to `code`, it will work for normal VSCode too.

```Dockerfile
ENV SERVICE_URL=https://marketplace.mycompany.com/_apis/public/gallery \
    ITEM_URL=https://marketplace.mycompany.com/items

RUN code-server --install-extension Shan.code-settings-sync

RUN echo "{}" >> /root/.local/share/code-server/User/settings.json

RUN tmp=$(mktemp) && \
      jq -M '. +{ \
      "sync.syncExtensions": true,\
      "sync.autoDownload": true,\
      "sync.gist":"abc123def987",\
      "telemetry.enableTelemetry": false }' /root/.local/share/code-server/User/settings.json > "$tmp" && mv "$tmp" /root/.local/share/code-server/User/settings.json

RUN echo "{}" >> /root/.local/share/code-server/User/syncLocalSettings.json

RUN tmp=$(mktemp) && \
      jq -M '. +{ "downloadPublicGist": true }' /root/.local/share/code-server/User/syncLocalSettings.json > "$tmp" && mv "$tmp" /root/.local/share/code-server/User/syncLocalSettings.json
```

Basically creating VSCode `settings.json` and adding gist ID to it using `jq` and then creating `syncLocalSettings.json` config with `"downloadPublicGist": true`.
When VSCode (code-server) is opened first time, settings download gets successfully started.
If you don't want auto-download to start, change `"sync.autoDownload": true` to `false`

Now I only need to find a way to initialize VSCode first run bootstrap through command line /bash

**Comment by sr229** [comment]
Really a interesting use case but we'll see what the team thinks.

**Comment by Zireael** [comment]
Thanks, I think `code-server` strong use case is automated installations for mass user provision (something like jupyterlab + jupyterhub in docker swarm or kubernetes).

It would be cool if I could supply a gist ID or [code-settings-sync](https://github.com/shanalikhan/code-settings-sync) file during code-server dockerimage install and the entire environment (installed plugins, themes, VSCode settings) got set up and ready for users. Jupyter is used like that in schools/unis, during conferences, bootcamps, for company team environments and so on. 

Every time you wanted to provide entry level users with instant complete coding environment for a particular task, this would be the use case.

**Comment by sr229** [comment]
To be honest this seems like a good use case for automated user setup for enterprises as well.

**Comment by jrderuiter** [comment]
At the moment, we're bootstrapping the config using /etc/skel when a user is first created (which happens when a user is provisioned to the machine). Extensions are deployed from S3 using a custom sync script, as we are unable to install extensions at runtime due to a http proxy that's not being picked up by code-server.

**Comment by sr229** [comment]
@jrderuiter sounds like a bug, would you be fine opening that as a seperate issue?

**Comment by nhooyr** [comment]
@code-asher is this possible?

**Comment by nhooyr** [comment]
Could you clarify why `code-server --install-extension xxx.yyy` does not work for you?

I'm not sure if there is a problem here.
